src/main/java/org/broadinstitute/hellbender/tools/exome/sexgenotyper/TargetCoverageSexGenotypeCalculator.java

```diff
@@ -28,141 +32,335 @@
-                    " read count collection.");
-        }
-
-        final ImmutablePair<ReadCountCollection, List<Target>> processedReadCountsAndTargets =
-                processReadCountsAndTargets(rawReadCounts, targetList);
-        processedReadCounts = processedReadCountsAndTargets.left;
-        final List<Target> processedTargetList = processedReadCountsAndTargets.right;
+        /* keep genotyping targets that exist in the read count collection */
+        final Set<Target> readCountTargets = new HashSet<>(readCounts.targets());
+        final List<Target> mutualTargets = genotypingTargets.stream()
+                .filter(readCountTargets::contains)
+                .collect(Collectors.toList());
 
         /* annotate targets with germline ploidy data */
         final GermlinePloidyAnnotatedTargetCollection ploidyAnnots =
-                new GermlinePloidyAnnotatedTargetCollection(contigPloidyAnnotsList, processedTargetList);
+                new GermlinePloidyAnnotatedTargetCollection(contigPloidyAnnotsList, mutualTargets);
 
         /* populate lists and maps */
         autosomalTargetList = ploidyAnnots.getAutosomalTargetList();
         allosomalTargetList = ploidyAnnots.getAllosomalTargetList();
         sexGenotypeIdentifiers = contigPloidyAnnotsList.get(0).getGenotypesSet();
 
-        if (allosomalTargetList.isEmpty()) {
-            throw new UserException.BadInput("No allosomal targets could be identified. Please ensure (1) the " +
-                    "read count collections covers some allosomal targets, and (2) allosomal targets are properly " +
-                    "annotated");
+        /* autosomal targets have the same ploidy for all ploidy classes (this is asserted earlier); so pick the first one */
+        autosomalTargetPloidies = getPloidyArray(sexGenotypeIdentifiers.iterator().next(), autosomalTargetList, ploidyAnnots);
+
+        /* calculate mean autosomal ploidy */
+        meanAutosomalPloidy = getAutosomalMeanPloidy(autosomalTargetList, autosomalTargetPloidies);
+
+        /* allosomal targets may have different ploidies for different ploidy classes */
+        allosomalTargetPloidies = getAllosomalPloidyMap(allosomalTargetList, sexGenotypeIdentifiers, ploidyAnnots);
+
+        /* initialize the multiplicative biases and masks */
+        autosomalTargetBiases = new double[autosomalTargetList.size()];
+        Arrays.fill(autosomalTargetBiases, 1.0);
+        autosomalTargetMasks = new int[autosomalTargetList.size()];
+        Arrays.fill(autosomalTargetMasks, 1);
+        allosomalTargetBiases = new double[allosomalTargetList.size()];
+        Arrays.fill(allosomalTargetBiases, 1.0);
+        allosomalTargetMasks = new int[allosomalTargetList.size()];
+        Arrays.fill(allosomalTargetMasks, 1);
+
+        /* take into account bait count bias */
+        if (processBaitCountAnnotations(autosomalTargetList, autosomalTargetBiases, autosomalTargetMasks, "autosomal")) {
```

Comment:
This control flow is a little idiosyncratic and does not save much compute.  I'd just make `processBaitCountAnnotations` a `void` method and remove the if statement.